pkg/redisx/pool.go

```diff
@@ -0,0 +1,45 @@
+package redisx
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/aquasecurity/harbor-scanner-trivy/pkg/etc"
+
+	"github.com/gomodule/redigo/redis"
+	log "github.com/sirupsen/logrus"
+)
+
+// NewPool constructs a redis.Pool with the specified configuration.
+//
+// The URI scheme currently supports connections to a standalone Redis server,
+// i.e. `redis://user:password@host:port/db-number`.
+func NewPool(config etc.RedisPool) (pool *redis.Pool, err error) {
+	configURL, err := url.Parse(config.URL)
+	if err != nil {
+		err = fmt.Errorf("invalid redis URL: %s", err)
+		return
+	}
+
+	switch configURL.Scheme {
+	case "redis":
```

Comment:
Here we'll be able to support redis-sentinel schema without changing the client code